### ISSUE TITLE
[css-highlight-api] Exposing shadow DOM highlights in highlightsFromPoint()

### ISSUE THREAD

**Issue by dandclark** [w3c/csswg-drafts#7766]
The `highlightsFromPoint()` function, discussed [here](https://github.com/w3c/csswg-drafts/issues/7513#issuecomment-1248320696), should probably not return highlights in shadow DOMs by default. But, there should still be a way for authors to allow interaction with highlights in shadows using `highlightsFromPoint()`.

I can think of two ways to have this function expose shadow DOM highlights.

The first is to handle it like `getInnerHTML()` does for Declarative Shadow DOM, in which it takes an `includeShadowRoots` option as well as a list of closed shadow roots. Highlights inside those shadows can be returned without breaking encapsulation, sinc the caller has proven that they already know about them: https://web.dev/declarative-shadow-dom/#serialization. Highlights could do something similar:

```js
const hitTestResults = CSS.highlights.highlightsFromPoint({
  includeShadowRoots: true,
  closedRoots: [shadowRoot1, shadowRoot2, ...]
});
```

Alternatively, we could put `highlightsFromPoint()` on [DocumentOrShadowRoot](https://dom.spec.whatwg.org/#mixin-documentorshadowroot). When called on a Document, it would return only highlights in that document (not in its shadow roots). When called on a shadow root, it would return only highlights in that shadow, or perhaps highlights in that shadow plus highlights in the document (but not highlights in any nested shadow roots). This is analogous to how `elementsFromPoint()` works in Blink and Gecko today, though the [spec](https://drafts.csswg.org/cssom-view/#dom-document-elementsfrompoint) doesn't yet reflect that `elementsFromPoint()` is callable on shadow DOM and I see some cross-browser inconsistencies in which elements are returned.

**Comment by smaug----** [comment]
I could note there is https://github.com/mfreed7/declarative-shadow-dom/issues/9

**Comment by stephanieyzhang** [comment]
Hi all, we propose putting `highlightsFromPoint()` on `CSS.highlights`, not `DocumentOrShadowRoot`, and adding an optional `options` dictionary parameter to the API. This dictionary has a key that maps to an array of `ShadowRoot` objects that can return highlights.
This approach is similar to how [`document.caretPositionFromPoint()` ](https://drafts.csswg.org/cssom-view/#dom-document-caretpositionfrompoint)and getHTML() interact with shadow DOM. Please let us know if there are any questions or concerns - thanks!

cc: @sanketj

**Comment by sanketj** [comment]
@dandclark @tabatkins Does Stephanie's [proposed syntax change](https://github.com/w3c/csswg-drafts/issues/7766#issuecomment-2257149661) seem reasonable to you?

**Comment by dandclark** [comment]
Yes, I think it's a reasonable change given the similarity to [document.caretPositionFromPoint()](https://drafts.csswg.org/cssom-view/#dom-document-caretpositionfrompoint).

**Comment by sanketj** [comment]
@astearns Do you think we can resolve async on the updates that @stephanieyzhang proposed [here](https://github.com/w3c/csswg-drafts/issues/7766#issuecomment-2257149661)? There are deltas compared to what the working group has resolved on previously, but I think they are relatively minor.

**Comment by astearns** [comment]
The CSSWG will automatically accept this resolution one week from now if no objections are raised here. Anyone can add an emoji to this comment to express support. If you do not support this resolution, please add a new comment.

Proposed Resolution: The `highlightsFromPoint()` method will be added to `CSS.highlights`, and have an optional `options` dictionary parameter matching the one from `caretPositionFromPoint()`

**Comment by astearns** [comment]
RESOLVED: The `highlightsFromPoint()` method will be added to `CSS.highlights`, and have an optional `options` dictionary parameter matching the one from `caretPositionFromPoint()`

**Comment by ffiori** [comment]
fixed with this PR https://github.com/w3c/csswg-drafts/pull/11507
